### ISSUE TITLE
Dynamic watermark for livestreaming

### ISSUE THREAD

**Issue by brunosiqueira** [igarape/copcast-admin#141]
To prevent screen­captures of video being viewed, all videos should also include dynamic watermarks during viewing, clearly labeling every single frame with the name of the viewer and the time of the view

**Comment by brunosiqueira** [comment]
We still have to check if this is doable with WebRTC

**Comment by jkosslyn** [comment]
I'd recommend prioritizing this as a key security feature

**Comment by brunosiqueira** [comment]
We should do this using html overlaying the video when playing. For exporting, ffmpeg watermarking will be necessary.

**Comment by martelli** [comment]
But wouldn't the overlaying allow for easy circumvention? A malicious user could easily grab the video link and save the file.

**Comment by brunosiqueira** [comment]
Yes it can. I thought about cost/benefit in this matter (I think we also
spoke about it in one meeting)
What would you suggest?

Em quinta-feira, 19 de novembro de 2015, Leandro Martelli <
notifications@github.com> escreveu:

> But wouldn't the overlaying allow for easy circumvention? A malicious user
> could easily grab the video link and save the file.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/igarape/copcast-admin/issues/141#issuecomment-158082403>
> .
>


-- 
Bruno Siqueira

**Comment by jkosslyn** [comment]
In our conversation the idea was to add the overlay dynamically on the server side

**Comment by martelli** [comment]
Dynamically re-encode and add the overlay on the fly, without generating 
a new file.

**Comment by brunosiqueira** [comment]
We have the overlay in the server side for copcast logo. When we discussed
placing also as a watermark the name of the user who was watching the video
( and/or the time stamp) we spoke about using html overlaying. This is for
playing only. Exporting would be all in the server side.

Em quinta-feira, 19 de novembro de 2015, jkosslyn <notifications@github.com>
escreveu:

> In our conversation the idea was to add the overlay dynamically on the
> server side
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/igarape/copcast-admin/issues/141#issuecomment-158095304>
> .
>


-- 
Bruno Siqueira

**Comment by brunosiqueira** [comment]
Would this take a lot of processing from the server? Imagining we will have
many users watching videos simultaneously?

Em quinta-feira, 19 de novembro de 2015, Leandro Martelli <
notifications@github.com> escreveu:

> Dynamically re-encode and add the overlay on the fly, without generating
> a new file.
>
>
> On 11/19/2015 01:43 PM, Bruno Siqueira wrote:
> > Yes it can. I thought about cost/benefit in this matter (I think we also
> > spoke about it in one meeting)
> > What would you suggest?
> >
> > Em quinta-feira, 19 de novembro de 2015, Leandro Martelli <
> > notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>> escreveu:
> >
> > > But wouldn't the overlaying allow for easy circumvention? A
> > malicious user
> > > could easily grab the video link and save the file.
> > >
> > > —
> > > Reply to this email directly or view it on GitHub
> > >
> > <
> https://github.com/igarape/copcast-admin/issues/141#issuecomment-158082403
> >
> > > .
> > >
> >
> >
> > --
> > Bruno Siqueira
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/igarape/copcast-admin/issues/141#issuecomment-158094751
> >.
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/igarape/copcast-admin/issues/141#issuecomment-158097121>
> .
>


-- 
Bruno Siqueira

**Comment by martelli** [comment]
Would impose some load, sure. But I think that's the way to go, 
otherwise we would leave a bypass open.
And we can always add more servers to distribute the load.

**Comment by brunosiqueira** [comment]
Agreed. Let's do it like this then.

Em quinta-feira, 19 de novembro de 2015, Leandro Martelli <
notifications@github.com> escreveu:

> Would impose some load, sure. But I think that's the way to go,
> otherwise we would leave a bypass open.
> And we can always add more servers to distribute the load.
>
> On 11/19/2015 01:51 PM, Bruno Siqueira wrote:
> > Would this take a lot of processing from the server? Imagining we will
> > have
> > many users watching videos simultaneously?
> >
> > Em quinta-feira, 19 de novembro de 2015, Leandro Martelli <
> > notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>> escreveu:
> >
> > > Dynamically re-encode and add the overlay on the fly, without
> generating
> > > a new file.
> > >
> > >
> > > On 11/19/2015 01:43 PM, Bruno Siqueira wrote:
> > > > Yes it can. I thought about cost/benefit in this matter (I think
> > we also
> > > > spoke about it in one meeting)
> > > > What would you suggest?
> > > >
> > > > Em quinta-feira, 19 de novembro de 2015, Leandro Martelli <
> > > > notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>
> > > <javascript:_e(%7B%7D,'cvml','notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>');>> escreveu:
> > > >
> > > > > But wouldn't the overlaying allow for easy circumvention? A
> > > > malicious user
> > > > > could easily grab the video link and save the file.
> > > > >
> > > > > —
> > > > > Reply to this email directly or view it on GitHub
> > > > >
> > > > <
> > >
> >
> https://github.com/igarape/copcast-admin/issues/141#issuecomment-158082403
> > > >
> > > > > .
> > > > >
> > > >
> > > >
> > > > --
> > > > Bruno Siqueira
> > > >
> > > > —
> > > > Reply to this email directly or view it on GitHub
> > > > <
> > >
> >
> https://github.com/igarape/copcast-admin/issues/141#issuecomment-158094751
> > > >.
> > > >
> > >
> > > —
> > > Reply to this email directly or view it on GitHub
> > >
> > <
> https://github.com/igarape/copcast-admin/issues/141#issuecomment-158097121
> >
> > > .
> > >
> >
> >
> > --
> > Bruno Siqueira
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/igarape/copcast-admin/issues/141#issuecomment-158097811
> >.
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/igarape/copcast-admin/issues/141#issuecomment-158099261>
> .
>


-- 
Bruno Siqueira
